### PR TITLE
Setup ActivityPreset and UserActivity

### LINKED ISSUE #18: Create an entity for user activity (swiping)
Create an entity for user activity #3, where the information of the activity and of users associated to that activity is stored.

Time estimate: 1h

### LINKED ISSUE #21: Create an entity for activity preset (swiping)
Create an entity for activity that contains preset information about an activity #3.

Time estimate: 1h

### PR DESCRIPTION
**Pull request by AleksandarRistic99**
closes #18
closes #21

### CODE REVIEW

**Comment by AleksandarRistic99** [comment]
Also added support classes and Repositories.

-Made UserStatus for activity --> UserDecisionStatus (mentioned in presentation to change the name)

-Made UserDecisionStatus to be an entity in a table, otherwise no relationship between UserDecisionStatus and UserActivity was possible (maybe there is another solution to enable relationship but dont store UserDecisionStatus as a entity in a table ?)

**Review by SamuelBruegger — APPROVED** [review]
- ih de class UserDecisionStatus chönt meh sich öberlege de DecisionStatus zum ne enum z mache anstatt boolean, falls meh jemols wet funktione, wie superlike oder so ihbaue abr da isch halt eifach eh design frog, wo meh au spöter glaubs guet chöntet ändere

**Comment by AleksandarRistic99** [comment]
> * ih de class UserDecisionStatus chönt meh sich öberlege de DecisionStatus zum ne enum z mache anstatt boolean, falls meh jemols wet funktione, wie superlike oder so ihbaue abr da isch halt eifach eh design frog, wo meh au spöter glaubs guet chöntet ändere

Guete punkt, chame denn sicher no in zuekunft beachte, eventuell en enum mache wo TRUE, FALSE, und no wiiteri optione het wie SUPERLIKE

**Comment by AleksandarRistic99** [comment]
Han de vorschlag vom Samuel mal apasst, ich denk es schadet sicher ned wenn mer de code extendable halte

File: src/main/java/ch/uzh/ifi/hase/soprafs21/entities/ActivityPreset.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    private String activityName;
+
+    @Column(nullable = false)
+    private ActivityCategory activityCategory;
+
+    /**
+     * googlePOI was left out, since we have to discuss how the relationship shall be between
+     * the GooglePOI object and the activityPreset --> should one activity have multiple locations, since for example football can be played on multiple locations?
+     private GooglePOI googlePOI;
+     */
```
Comment (Taremeh) [inline src/main/java/ch/uzh/ifi/hase/soprafs21/entities/ActivityPreset.java]:
```suggestion
     private GooglePOI googlePOI;
```

I'd say that's alright. The GooglePOI object only contains minimal information for the API request (e.g. 'keyword = sushi' for an activity _eat at sushi restaurant_). There is no location saved in `GooglePOI`.

File: src/main/java/ch/uzh/ifi/hase/soprafs21/entities/ActivityPreset.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    @GeneratedValue
+    private Long id;
+
+    @Column(nullable = false, unique = true)
+    private String activityName;
+
+    @Column(nullable = false)
+    private ActivityCategory activityCategory;
+
+     private GooglePOI googlePOI;
```
Comment (Taremeh) [inline src/main/java/ch/uzh/ifi/hase/soprafs21/entities/ActivityPreset.java]:
```suggestion
    private GooglePOI googlePOI;
```

**Review by Taremeh — CHANGES_REQUESTED** [review]
- Good point regarding the `DecisionStatus`.
However, I would rename it to `SwipeStatus` since it's not directly clear what's meant by decision status. `SwipeStatus` would be clearer, also regarding SuperLike. 

- To simplify naming, I'd also suggest to call activities just `Activity` instead of `UserActivity`. (user activity sounds as if the user has created the activity which is not true)

- Regarding `GooglePOI`: that's all fine (see https://github.com/sopra-fs21-group-16/mth-server/pull/47#discussion_r606850667). Please implement the Spring Annotation and getters & setters. 

And btw. thx for deleting the outdated imports 🙏

**Comment by Taremeh** [comment]
⚠️  Still missing getters & setters for GooglePOI (https://github.com/sopra-fs21-group-16/mth-server/pull/47#discussion_r606850667)

**Pull request merged**